Sources/FunctionalTableData/CollectionView/AnyCollectionSection.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  AnyCollectionSection.swift
+//  FunctionalTableData
+//
+//  Created by Jason Kemp on 2021-10-28.
+//
+
+import UIKit
+
+struct AnyCollectionSection: Equatable, Hashable {
+	public static func ==(lhs: AnyCollectionSection, rhs: AnyCollectionSection) -> Bool {
+		return lhs.key == rhs.key
+	}
+
+	private var impl: CollectionSection
+	
+	public var key: String { impl.key }
+	public var items: [HashableCellConfigType]
+	public var supplementaries: [CollectionSupplementaryItemConfig] {
+		get { impl.supplementaries }
+		set { impl.supplementaries = newValue }
+	}
+			
+	public init(_ section: CollectionSection) {
+		items = section.items.map { AnyHashableConfig($0, sectionKey: section.key) }
+		impl = section
+	}
```

Comment:
Ah I see.

I tried something like that, but quickly ran into problems with diffing _and_ too many code changes at the call site.

For better or worse, we're stuck with `CellConfigType` as our primary type for declaring rows and items. So anything we adopt has to work with those. I can't make `AnyHashableConfig` replace `HashableCellConfigType` or `CellConfigType` anywhere because of the subtle differences in equality checking that classic FTD diffing has vs. diffable data sources.